src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -516,11 +516,21 @@ void HeapDumpDCmd::execute(DCmdSource source, TRAPS) {
     }
   }
 
+  // If the filename contains %p, it will be replaced by the pid.
+  char fname[JVM_MAXPATHLEN];
+  if (!Arguments::copy_expand_pid(_filename.value(), strlen(_filename.value()),
+                                  fname, JVM_MAXPATHLEN)) {
+    stringStream msg;
+    msg.print("Invalid file path name specified: %s", _filename.value());
+    THROW_MSG(vmSymbols::java_lang_IllegalArgumentException(), msg.base());
```

Comment:
Why throw? Why not just print an error message to the output() stream and return?